deps/rabbitmq_stomp/src/rabbit_stomp_processor.erl

```diff
@@ -1217,3 +1513,504 @@ maybe_apply_default_topic_exchange(Exchange, _DefaultTopicExchange) ->
+    %% io:format("Event: ~p~n", [Evt]),
+    %% io:format("QStates: ~p~n", [QStates0]),
+    case rabbit_queue_type:handle_event(QRef, Evt, QStates0) of
+        {ok, QState1, Actions} ->
+            %% io:format("ActionsEv ~p~n", [Actions]),
+            State1 = State#proc_state{queue_states = QState1},
+            State2 = handle_queue_actions(Actions, State1),
+            {ok, State2};
+        {eol, Actions} ->
+            State1 = handle_queue_actions(Actions, State),
+            State2 = handle_consuming_queue_down_or_eol(QRef, State1),
+            {ConfirmMXs, UC1} =
+                rabbit_confirms:remove_queue(QRef, State1#proc_state.unconfirmed),
+            %% Deleted queue is a special case.
+            %% Do not nack the "rejected" messages.
+            State3 = record_confirms(ConfirmMXs,
+                                     State2#proc_state{unconfirmed = UC1}),
+            {ok, State3#proc_state{queue_states = rabbit_queue_type:remove(QRef, QStates0)}};
+        {protocol_error, Type, Reason, ReasonArgs} = Error ->
+            log_error(Type, Reason, ReasonArgs),
+            {error, Error, State}
+    end.
+
+handle_queue_actions(Actions, #proc_state{} = State0) ->
+    %% io:format("Actions: ~p~n", [Actions]),
+    lists:foldl(
+      fun ({deliver, ConsumerTag, Ack, Msgs}, S) ->
+              deliver_to_client(ConsumerTag, Ack, Msgs, S);
+          ({settled, QRef, MsgSeqNos}, S0) ->
+              S = confirm(MsgSeqNos, QRef, S0),
+              send_confirms_and_nacks(S);
+          ({rejected, _QRef, MsgSeqNos}, S0) ->
+              {U, Rej} =
+              lists:foldr(
+                fun(SeqNo, {U1, Acc}) ->
+                        case rabbit_confirms:reject(SeqNo, U1) of
+                            {ok, MX, U2} ->
+                                {U2, [MX | Acc]};
+                            {error, not_found} ->
+                                {U1, Acc}
+                        end
+                end, {S0#proc_state.unconfirmed, []}, MsgSeqNos),
+              S = S0#proc_state{unconfirmed = U},
+              %% Don't send anything, no nacks in STOMP
+              record_rejects(Rej, S);
+          ({queue_down, QRef}, S0) ->
+              handle_consuming_queue_down_or_eol(QRef, S0);
+          %% TODO: I have no idea about the scope of credit_flow
+          ({block, QName}, S0) ->
+              credit_flow:block(QName),
+              S0;
+          ({unblock, QName}, S0) ->
+              credit_flow:unblock(QName),
+              S0;
+          %% TODO: in rabbit_channel there code for handling
+          %% send_drained and send_credit_reply
+          %% I'm doing catch all here to not crash?
+          (_, S0) ->
+              S0
+      end, State0, Actions).
+
+
+
+parse_endpoint(undefined) ->
+    parse_endpoint("/queue");
+parse_endpoint(Destination) when is_binary(Destination) ->
+    parse_endpoint(unicode:characters_to_list(Destination));
```

Comment:
Where is this code tested? If it's not tested, please add tests.

Reply:
parse_endpoint called on each SEND and SUBSCRIBE. Should be many dozens hits across our test suites

Reply:
I observe 0 hits for these 2 clauses ☹️

<img width="768" alt="Screenshot 2023-09-06 at 08 52 17" src="https://github.com/rabbitmq/rabbitmq-server/assets/12648310/a1e63115-518c-4bdd-a3d6-b0e918773e2e">